paywall/src/paywall-script/index.ts

```diff
@@ -1,14 +1,31 @@
 import Postmate from 'postmate'
+import {
+  CheckoutEvents,
+  UserInfo,
+  TransactionInfo,
+} from '../../../unlock-app/src/hooks/useCheckoutCommunication'
 
 declare let __ENVIRONMENT_VARIABLES__: { unlockAppUrl: string }
 
 const { unlockAppUrl } = __ENVIRONMENT_VARIABLES__
 const rawConfig = (window as any).unlockProtocolConfig
-const encodedConfig = encodeURI(JSON.stringify(rawConfig))
+const encodedConfig = encodeURIComponent(JSON.stringify(rawConfig))
```

Comment:
This slight difference caused the config to not always encode in a reversible way.